examples/timeSeries-hand-gestures/sketch.js

```diff
@@ -0,0 +1,217 @@
+  
+  // setup the timeseries neural network
+  let options = {
+    outputs: ['label'],
+    task: 'classification',
+    dataModality: 'spatial',
+    debug: 'true',
+    learningRate: 0.001,
+  };
+  model = ml5.timeSeries(options);
+
+}
+
+function draw() {
+  // draw video on frame
+  image(video, 0, 0, width, height);
+  
+  drawPredictedWord();
+
+  // if hands are found then start recording
+  if(hands.length>0 && recording_finished == false){
+    if (sequence.length <= seqlength){
+      // get coordinates from hands (21 points)
+      handpoints = drawPoints();
+      sequence.push(handpoints);
+
+    // once sequence reaches the seqlength, add sequence as just one X value
+    } else if (sequence.length>0){
+      // get the training word from the input box
+      let train_word = nameField.value()
+
+      // if there is a word currently in the box then add data with that label
+      if (train_word.length > 0){
+        // add data to the model
+        let target = {label:train_word}
+        model.addData(sequence, target);
+        trainingWordsUpdate()
+
+      // if there is no word in the box then classify instead
+      } else {
+        // classify the data
+        model.classify(sequence, gotResults);
+      }
+
+      // reset the sequence
+      sequence = [];
+      recording_finished = true;
+    }
+
+  // can only record again when hand is out of frame
+  } else {
+    if (hands.length == 0){
+      recording_finished = false;
+    }
+  }
+}
+
+function drawPoints(){
+  let handpoints = []
+  // iterate through both hands
+  for (let i = 0; i < hands.length; i++) {
+    let hand = hands[i];
+    for (let j = 0; j < hand.keypoints.length; j++) {
+      // access the keypoints in the hand
+      let keypoint = hand.keypoints[j];
+      handpoints.push(keypoint.x,keypoint.y)
+
+      fill(0, 255, 0);
+      noStroke();
+      circle(keypoint.x, keypoint.y, 5);
+    }
+  }
+  // assign to a different variable before clearing
+  const output = handpoints;
+  handpoints = []; 
+  
+  return output;
+}
+
+// Callback function for when handPose outputs data
+function gotHands(results) {
+  // save the output to the hands variable
+  hands = results;
+}
+
+function keyPressed(){
+  if (key == 's'){
+    model.save('hello');
+  }
+  if (key == 'z'){
+    model.saveData();
+  }
+
+  if (key == 't'){
+    model.normalizeData();
+    let options = {
+      epochs: 100
+    }
+    model.train(options,whileTraining,finishedTraining);
+  }
+}
+
+function trainModelAndSave(){
+  model.normalizeData();
+  let options = {
+    epochs: 100
+  }
+  model.train(options,whileTraining,finishedTraining);
+  nameField.value('')
+}
+
+function whileTraining(epoch) {
+  console.log(epoch);
+}
+
+function finishedTraining() {
+  console.log('finished training.');
+  model.save('model');
+}
+
+function gotResults(results){
+  predicted_word = results[0].label
+  console.log(predicted_word)
+  text(predicted_word, 200,200)
+}
+
+function UI(){
```

Comment:
Hi professor, I did this for both the mouse gesture and the weather predictor, they now only have buttons but have helpful texts when pressed. For the hand gestures, specifically for the train and save, I think there is a way to sort of minimize the text at the bottom, however, I patterned the UI from the train and save of the Neural network and it might be helpful if there is a consistency between the two. I am open to both and am currently drafting code for alternate UI